### PR TITLE
drivers: sensor: SCD4x Add driver

### PR DESCRIPTION
**Pull request by jfaeh**
This adds support for Sensirion's SCD4x co2 sensor.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Please try to address the CI failures.

**Review by MaureenHelm — DISMISSED** [review]

**Comment by jfaeh** [comment]
Resolved merge conflict in i2c.dtsi

**Comment by jfaeh** [comment]
@MaureenHelm @ubieda @jeppenodgaard resolved merge conflict once again

**Review by MaureenHelm — APPROVED** [review]

File: tests/drivers/build_all/sensor/i2c.dtsi [hunk 1]
```diff
@@ -1131,3 +1131,9 @@ test_i2c_sts4x: sts4x@9d {
 	reg = <0x99>;
 	repeatability = <2>;
 };
```
Comment (kartben) [inline tests/drivers/build_all/sensor/i2c.dtsi:1133]: Looks like the reg property of the previous sensor binding is wrong, but obviously not your fault :)

**Review by kartben — APPROVED** [review]

**Pull request merged**